R/Rules-validity.R

```diff
@@ -48,6 +48,35 @@ v_next_best_ncrm <- function(object) {
   v$result()
 }
 
+#' @describeIn v_next_best validates that the [`NextBestNCRMLoss`] object
+#'   contains valid objects.
+v_next_best_ncrm_loss <- function(object) {
+  v <- Validate()
+  v$check(
+    test_probability_range(object@target_int),
+    "target_int has to be a probability range"
+  )
+  v$check(
+    test_probability_range(object@overdose_int),
+    "overdose_int has to be a probability range"
+  )
+  if (object@unacceptable_int[1] != 1) {
+    v$check(
+      test_probability_range(object@unacceptable_int),
+      "unacceptable_int has to be a probability range"
+    )
+  }
+  v$check(
+    test_probability(object@max_overdose_prob, bounds_closed = FALSE),
+    "max_overdose_prob must be a probability value from (0, 1) interval"
+  )
+  v$check(
+    all(object@losses >= 0),
+    "losses has to be a numeric vector with all elements non-negative"
+  )
+  v$result()
+}
+
```

Comment:
I think this validation should be improved. For instance, 

1. what if condition in `if` clause is `FALSE`?
2. for `losses` please use `checkmate`
3. `overdose_int` - left bound in the interval is opened right? same for `unacceptable_int`.
4. lengths of `max_overdose_prob` and `losses` should also be verified.

Reply:
Sure; in particular `checkmate` provides a set of testing function `test_*` that we use to validate the objects.
For example 
```
test_numeric(object@weights, lower = 0, finite = TRUE, any.missing = FALSE, len = 2)
```
tests wether  `object@weights` is a numeric vector of length 2 without any missing values, with all values finite and greater than or equal to 0.

See more examples e.g. in `Model-validity.R` file.